compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -905,6 +923,21 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 #[derive(Debug, Copy, Clone)]
 enum OperandValueKind {
     Ref,
-    Immediate(abi::Scalar),
+    Immediate(ScalarOrZst),
```

Comment:
I wonder if this is best represented as a separate as a different variant `OperandValueKind::Zst`?

Reply:
I thought about that, but that loses the "same variants as `OperandValue`" property that I originally added this type for, as mentioned in the `value_kind` comments.

I think it's also a bit of a net loss in the places that lose it, since I need to look at it in pairs.  This form is nicer in `rvalue_creates_operand`, I'd say, rather than adding new arms for `ImmediateScalar`-and-`ImmediateZst` mismatches.

(Though I do wonder if maybe having a dedicated `OperandValue::Zst` could make sense, rather than [needing to pass around a useless `Bx::Value` for those cases](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_codegen_ssa/mir/operand/struct.OperandRef.html#method.new_zst) at all.)

Reply:
are zst poisons a thing? that would be a reason to carry around a `Bx::Value` I guess?

Reply:
We certainly do pass around the `Bx::Value`s for them, but they also don't seem to end up [in even the `-C no-prepopulate-passes -Z mir-opt-level=0` IR](https://rust.godbolt.org/z/qWT6YvKjh), so 🤷

Reply:
Looks like they're a thing, because opt and llc at least don't complain about LLVM IR like `ret {} poison`: <https://llvm.godbolt.org/z/ccYYMKjYb> -- not that that's how we return ZSTs in Rust.